lib/at_cmd_parser/at_utils.h

```diff
@@ -242,6 +242,30 @@ static inline bool is_command(const char *str)
 	return false;
 }
 
+/**
+ * @brief Check if a string is a beginning of an AT CLAC response
+ *
+ * This function will check if the string is a CLAC response prefix.
+ * Valid prefixes: AT+ and AT%, except AT%X
+ * @param[in] str String to examine
+ *
+ * @retval true  If the string is a CLAC response
+ * @retval false Otherwise
+ */
+static inline bool is_clac(const char *str)
+{
+	if (strlen(str) < 3) {
+		return false;
+	}
+
+	if ((toupper((int)str[0]) == 'A') &&
+		(toupper((int)str[1]) == 'T') &&
+		((toupper((int)str[2]) == '+') ||
+		((toupper((int)str[2]) == '%') &&  (toupper((int)str[3]) != 'X')))) {
+		return true;
+	}
```

Comment:
Perhaps tweak the indentation to make it a bit more readable, something like:
```suggestion
	if ((toupper((int)str[0]) == 'A') &&
	    (toupper((int)str[1]) == 'T') &&
	    ((toupper((int)str[2]) == '+') ||
	     ((toupper((int)str[2]) == '%') &&
	      (toupper((int)str[3]) != 'X')))) {
		return true;
	}
```
Still not the most readable `if` but I don't have a better idea for now.

Reply:
You could turn the conditions around like this:

```
if (toupper((int)str[0]) != 'A' && toupper((int)str[1]) != 'T') return false; // No AT cmd
if (toupper((int)str[2]) != '+' && toupper((int)str[2]) != '%') return false; // Neither AT+ nor AT%
if (toupper((int)str[2]) == '%' && toupper((int)str[3]) == 'X') return false; // ignore AT%X
return true;
```

Reply:
Just note that Zephyr (and thus NCS) coding guideline requires surrounding `{ }` even for single-line blocks, so it'd rather be:
```
if (toupper((int)str[0]) != 'A' && toupper((int)str[1]) != 'T') {
	return false;
}
```
etc.